components/Accordion/react/Accordion.tsx

```diff
@@ -29,22 +69,82 @@ export const Accordion: React.FC<
   titleClassName,
   descriptionClassName,
   fullWidthContent,
+  open,
+  onClickSummary,
+  onToggle = () => {},
   ...rest
 }) => {
-  const details = React.useRef(null)
-  const content = React.useRef(null)
+  const [isOpen, setIsOpen] = useState(open)
+  const details = React.useRef<HTMLDetailsElement>(null)
+  const summary = React.useRef<HTMLDivElement>(null)
+  const content = React.useRef<HTMLDivElement>(null)
+
+  // Synchronize internal state with the `open` prop
+  useEffect(() => {
+    setIsOpen(open)
+  }, [open])
+
   React.useEffect(() => {
     if (details.current && content.current) {
       new DetailsAnimation(details.current, content.current)
     }
   }, [])
+
+  useEffect(() => {
+    const summaryElement = summary?.current
+
+    if (summaryElement) {
+      const handleSummaryClick = (event: MouseEvent) => {
+        // Stop the native event
+        event.preventDefault()
+        event.stopPropagation()
+
+        let onClickRet: boolean | undefined = true
+        if (onClickSummary) {
+          onClickRet = onClickSummary(event)
+        }
+
+        // TODO: Clone the original event then check propagation.
```

Comment:
Is this an open todo?

Reply:
It is. The ideal code would clone the original event and link stopPropagation. For now, that's no ideal for the time we have to finish and it's not a blocker.

Reply:
is there an issue to link to track that?

Reply:
No. Actually I'm not sure we need it until we know that we will need it. In other words, it's not absolutely required but I wanted to leave a note in there in case someone wonders why stopPropagation doesn't work.